tree.go

```diff
@@ -569,13 +568,6 @@ func (n *InternalNode) Commitment() *Point {
 }
 
 func (n *InternalNode) Commit() *Point {
-	// If we're at the first or second layer, we do the Commit() in parallel
-	// leveraging available cores.
-	// If we're in further layers, we do single-goroutine Commit() since the top
-	// two layers already created enough goroutines doing downstream work.
-	if n.depth <= 1 {
-		return n.commitParallel()
-	}
```

Comment:
Now that I did other optimizations, still looking at benchmarks I noticed this parallelization started to hurt.

Things were much faster now, and we're doing less allocations (so less GC pressure). At this point I noticed this was putting a lot of pressure on the scheduler of goroutines and doing some extra heap escaping that affected performance.

This isn't a mere "let's simplify now that we're faster"... this was literally slowing things down by a 2-digit %.
The faster you go, the more these "parallelization overhead" start to count.

I think this is great... we're faster using a single core!